### PR TITLE
refactor: replace a few any-s with proper types

### PR DESCRIPTION
**Pull request by miniak**
#### Description of Change
Improves type safety.

#### Checklist
- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

#### Release Notes
Notes: no-notes

### CODE REVIEW

File: lib/common/api/deprecate.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ const deprecate: ElectronInternal.DeprecationUtil = {
     return function (this: any) {
       warn();
       fn.apply(this, arguments);
-    };
+    } as any;
```
Comment (nornagon) [inline lib/common/api/deprecate.ts:46]: does this work? same effect i suppose from the outside, but it's a bit better documentation
```suggestion
    } as any as typeof fn;
```

File: lib/renderer/window-setup.ts [hunk 2]
```diff
@@ -255,13 +255,13 @@ export const windowSetup = (
...
     // Make the browser window or guest view emit "new-window" event.
-    (window as any).open = function (url?: string, frameName?: string, features?: string) {
+    window.open = function (url?: string, frameName?: string, features?: string) {
       if (url != null && url !== '') {
         url = resolveURL(url, location.href);
       }
       const guestId = ipcRendererInternal.sendSync('ELECTRON_GUEST_WINDOW_MANAGER_WINDOW_OPEN', url, toString(frameName), toString(features));
       if (guestId != null) {
-        return getOrCreateProxy(guestId);
+        return getOrCreateProxy(guestId) as any as WindowProxy;
```
Comment (nornagon) [inline lib/renderer/window-setup.ts:264]: can this not return `WindowProxy` directly?

Comment (miniak) [reply lib/renderer/window-setup.ts:264]:
```
Type 'SafelyBoundBrowserWindowProxy' is missing the following properties from type 'Window': applicationCache, clientInformation, customElements, defaultStatus, and 237 more.ts(2740)
```

File: lib/common/api/deprecate.ts [hunk 3]
```diff
@@ -52,7 +52,7 @@ const deprecate: ElectronInternal.DeprecationUtil = {
     return function (this: any) {
       warn();
       return fn.apply(this, arguments);
-    };
+    } as any;
```
Comment (nornagon) [inline lib/common/api/deprecate.ts:55]: here too?

**Review by nornagon — APPROVED** [review]

**Review by alexeykuzmin — APPROVED** [review]

**Pull request merged**